test/python/test_resources_api.py

```diff
@@ -8,35 +8,215 @@
+        """Test case for 204 status response on /resources/{account}/{kind}/{identifier}
+        204 - permission check successful
+        """
+        response = self.api.get_resource_with_http_info(
+            self.account,
+            'variable',
+            'testSecret',
+            privilege="update",
+            check=True
+        )
+        self.assertEqual(response[1], 204)
+        self.assertEqual(response[0], "")
 
-    def test_get_resources(self):
-        """Test case for get_resources
+    def test_get_resource_400(self):
+        """Test case for 400 status response on /resources/{account}/{kind}/{identifier} endpoint
+        400 - a required URI parameter is malformed
+        """
+        try:
+            self.api.get_resource(self.account, NULL_BYTE, 'testSecret')
+            self.fail("No Error! Expected 400 status response.")
+        except openapi_client.ApiException as exception:
+            self.assertEqual(exception.status, 400)
+
+    def test_get_resource_401(self):
+        """Test case for 401 status response on /resources/{account}/{kind}/{identifier} endpoint
+        401 - unauthorized request
+        """
+        try:
+            self.bad_auth_api.get_resource(self.account, 'variable', 'testSecret')
+            self.fail("No Error! Expected 401 status response.")
+        except openapi_client.ApiException as exception:
+            self.assertEqual(exception.status, 401)
+
+    def test_get_resource_404a(self):
+        """Test case for 404 status response on /resources/{account}/{kind}/{identifier} endpoint
+        404 - the requested resource does not exist
+        """
+        try:
+            self.api.get_resource(self.account, 'variable', 'fakeVariable')
+            self.fail("No Error! Expected 404 status response.")
+        except openapi_client.ApiException as exception:
+            self.assertEqual(exception.status, 404)
 
-        Lists resources within an organization account.
+    def test_get_resource_404b(self):
+        """Test case for 404 status response on /resources/{account}/{kind}/{identifier} endpoint
+        404 - the authenticated user lacks the necessary privilege
+        Conjur docs say this should return 403 - Conjur api feature tests say otherwise
         """
-        resources = self.api.get_resources(self.account)
+        # create and authenticate alice, a user with no resource permissions
+        policy_api = openapi_client.PoliciesApi(self.client)
 
-        self.assertIsInstance(resources, list)
-        for resource in resources:
+        alice_info = policy_api.modify_policy(self.account, 'root', POLICY_ALICE)
+        alice_api_key = alice_info["created_roles"]["dev:user:alice"]["api_key"]
+
+        config = api_config.get_api_config()
+        config.username = "alice"
+        alice_client = openapi_client.ApiClient(config)
+        authn_api = openapi_client.AuthnApi(alice_client)
+        alice_authn_token = authn_api.authenticate(
```

Comment:
I would move the logic for getting a new client authenticated with user `alice` into another function, just in the name of keeping the test methods as small and focused on the actual test content as possible.